tests/integration/Core/Repository/ContentService/CopyNonTranslatableFieldsFromPublishedVersionTest.php

```diff
@@ -0,0 +1,156 @@
+
+        // Updating non-translatable field in eng-US language (allowed) and publishing it
+        $engContent = $contentService->createContentDraft($publishedContent->contentInfo);
+
+        $contentUpdateStruct = new ContentUpdateStruct([
+            'initialLanguageCode' => self::ENG_US,
+            'fields' => $contentDraft->getFields(),
+        ]);
+
+        $expectedBodyValue = 'Nontranslatable value';
+        $contentUpdateStruct->setField('title', 'Title v2', self::ENG_US);
+        $contentUpdateStruct->setField('body', $expectedBodyValue, self::ENG_US);
+
+        $engContent = $contentService->updateContent($engContent->getVersionInfo(), $contentUpdateStruct);
+        $contentService->publishVersion($engContent->getVersionInfo());
+
+        // Publishing ger-DE draft with the empty non-translatable field
+        $contentService->publishVersion($gerContent->getVersionInfo());
+
+        // Loading main content
+        $mainPublishedContent = $contentService->loadContent($engContent->id);
+        $bodyFieldValue = $mainPublishedContent->getField('body')->getValue();
+
+        self::assertSame($expectedBodyValue, $bodyFieldValue->text);
+    }
+
+    private function createNonTranslatableContentType(): void
+    {
+        $permissionResolver = self::getPermissionResolver();
+        $contentTypeService = self::getContentTypeService();
+
+        $typeCreate = $contentTypeService->newContentTypeCreateStruct(self::CONTENT_TYPE_IDENTIFIER);
+
+        $typeCreate->mainLanguageCode = 'eng-GB';
+        $typeCreate->remoteId = '1234567890abcdef';
+        $typeCreate->urlAliasSchema = '<title>';
+        $typeCreate->nameSchema = '<title>';
+        $typeCreate->names = [
+            'eng-GB' => 'Nontranslatable content type',
+        ];
+        $typeCreate->descriptions = [
+            'eng-GB' => '',
+        ];
+        $typeCreate->creatorId = $permissionResolver->getCurrentUserReference()->getUserId();
+        $typeCreate->creationDate = new DateTime();
+
+        $titleFieldCreate = $contentTypeService->newFieldDefinitionCreateStruct('title', 'ezstring');
+        $titleFieldCreate->names = [
+            'eng-GB' => 'Title',
+        ];
+        $titleFieldCreate->descriptions = [
+            'eng-GB' => 'Title',
+        ];
+        $titleFieldCreate->fieldGroup = 'content';
+        $titleFieldCreate->position = 1;
+        $titleFieldCreate->isTranslatable = true;
+        $titleFieldCreate->isRequired = true;
+        $titleFieldCreate->isInfoCollector = false;
+        $titleFieldCreate->validatorConfiguration = [
+            'StringLengthValidator' => [
+                'minStringLength' => 0,
+                'maxStringLength' => 0,
+            ],
+        ];
+        $titleFieldCreate->fieldSettings = [];
+        $titleFieldCreate->isSearchable = true;
+        $titleFieldCreate->defaultValue = 'default title';
+
+        $typeCreate->addFieldDefinition($titleFieldCreate);
```

Comment:
This fragment feels still a bit too similar to L126-148. What if we extracted variable parts and created separate method for that, so the entire block L102-148 is more compact?

```php
        $fieldDefinitionPosition = 1;
        $typeCreate->addFieldDefinition(
            $this->buildFieldDefinitionCreateStructForNonTranslatableContentType(
                $fieldDefinitionPosition,
                'title',
                ['eng-GB' => 'Title'],
                true,
                true,
                'default title'
            )
        );

        $typeCreate->addFieldDefinition(
            $this->buildFieldDefinitionCreateStructForNonTranslatableContentType(
                ++$fieldDefinitionPosition,
                'body',
                ['eng-GB' => 'Body'],
                false,
                false
            )
        );
```
and the method itself:
```php
// private const TEXT_LINE_FIELD_TYPE_IDENTIFIER = 'ezstring';

    /**
     * @param array<string, string> $names
     */
    private function buildFieldDefinitionCreateStructForNonTranslatableContentType(
        int $position,
        string $fieldIdentifier,
        array $names,
        bool $isTranslatable,
        bool $isRequired,
        ?string $defaultValue = null
    ): FieldDefinitionCreateStruct {
        $contentTypeService = self::getContentTypeService();

        $fieldDefinitionCreateStruct = $contentTypeService->newFieldDefinitionCreateStruct(
            $fieldIdentifier,
            self::TEXT_LINE_FIELD_TYPE_IDENTIFIER
        );

        $fieldDefinitionCreateStruct->names = $names;
        $fieldDefinitionCreateStruct->descriptions = $names;
        $fieldDefinitionCreateStruct->fieldGroup = 'content';
        $fieldDefinitionCreateStruct->position = $position;
        $fieldDefinitionCreateStruct->isTranslatable = $isTranslatable;
        $fieldDefinitionCreateStruct->isRequired = $isRequired;
        $fieldDefinitionCreateStruct->isInfoCollector = false;
        $fieldDefinitionCreateStruct->validatorConfiguration = [
            'StringLengthValidator' => [
                'minStringLength' => 0,
                'maxStringLength' => 0,
            ],
        ];
        $fieldDefinitionCreateStruct->fieldSettings = [];
        $fieldDefinitionCreateStruct->isSearchable = true;
        $fieldDefinitionCreateStruct->defaultValue = $defaultValue;

        return $fieldDefinitionCreateStruct;
    }
```

I'm keeping common parts hardcoded. In the future either `RepositoryTestCase` or dedicated VO builder could have more flexible (less hardcoded) logic.

That should also conveniently get rid of SonarCloud Quality Gate failure.

Reply:
@alongosz sure, sounds good to me, please see https://github.com/ezsystems/ezplatform-kernel/pull/380/commits/7635e1413108327cba78236465620408490b55dd